controllers/post-controller.go

```diff
@@ -0,0 +1,65 @@
+package controllers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+
+	"github.com/julienschmidt/httprouter"
+
+	"github.com/ulumuri/rhododendron/models"
+)
+
+type Env struct {
+	DB models.Requester
+}
+
+func (env *Env) InsertPost(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	defer r.Body.Close()
+
+	post := &models.Post{}
+	err := json.NewDecoder(r.Body).Decode(post)
+	if err != nil {
+		respondWithJson(w, http.StatusBadRequest, err.Error())
+		return
+	}
+	_, err = env.DB.Insert(post)
+	if err != nil {
+		respondWithJson(w, http.StatusInternalServerError, err.Error())
```

Comment:
Don't forget to return in case of error.

BTW We could think about defining some error types / error response format for frontend.

Reply:
We should discuss it on discord later, just let me know whenever you have time.